File: py3/48-RotateImage.py
# 2022-04-22 22:29:10
# Runtime 36ms(68.8%)
# Memory 14.9mb(85.9%)

##
# 又作弊
###

class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        aux = [[i[j] for i in matrix][::-1] for j in range(len(matrix))]
        for i in range(len(matrix)):
            matrix[i] = aux[i]

File: cpp/48-RotateImage.cpp
// 2022-04-23 13:29:14
// Runtime 0ms(100.0%)
// Memory 6.9mb(56.8%)

class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        vector<vector<int>> aux;
        vector<int> line;
        int i, j;
        for (i = 0; i < matrix[0].size(); ++i) {
            line.clear();
            for (j = matrix.size() - 1; j > -1; --j) {
             line.push_back(matrix[j][i]);
            }
            aux.push_back(line);
        }
        matrix = aux;
    }
};
